src/resizing.jl

```diff
@@ -168,8 +168,16 @@ function restrict_indices(r::UnitRange)
 end
 
 # imresize
-imresize(original::AbstractArray, dim1, dimN...) = imresize(original, (dim1,dimN...))
+imresize(original::AbstractArray, dim1::T, dimN::T...) where T<:Union{Integer,AbstractUnitRange} = imresize(original, (dim1,dimN...))
+imresize(original::AbstractArray, short_size::AbstractArray{T,1}) where T <:Union{Integer,AbstractUnitRange} = imresize(original, (short_size...))
 
+function imresize(original::AbstractArray, ratio::Real)
+    # use ceil to avoid 0
+    new_size = ratio > 0 ? ceil.(Int, size(original).* ratio) : throw(ArgumentError("$ratio should be positive"))
```

Comment:
I tend to have a slight preference for
```julia
ratio > 0 || throw(ArgumentError("$ratio should be positive"))
new_size = ceil.(Int, size(original).* ratio)
```
rather than the trinary operator in such cases, but this is purely stylistic and not necessary to change.